src/components/TopNav.tsx

```diff
@@ -1,20 +1,322 @@
+                                <Link style={Links} to="/login">
+                                    SignIn
+                                </Link>
+                            }
+                        />
+                    </ListItemButton>
+                </ListItem>
+                <ListItem disablePadding>
+                    <ListItemButton sx={style.textCenter}>
+                        <ListItemText
+                            primary={
+                                <Link style={Links} to="/register">
+                                    SignUp
+                                </Link>
+                            }
+                        />
+                    </ListItemButton>
+                </ListItem>
+                <ListItem disablePadding>
+                    <ListItemButton sx={style.textCenter}>
+                        <ListItemText
+                            primary={
+                                <Link style={Links} to="/my-account/1">
+                                    Profile
+                                </Link>
+                            }
+                        />
+                    </ListItemButton>
+                </ListItem>
+            </List>
+        </Box>
+    );
+
+    const container =
+        window !== undefined ? () => window().document.body : undefined;
+    
     return (
-        <nav style={{ display: "flex", flexDirection: "column" }}>
-            <Link to="/">Home</Link>
-            <Link to="/posts/learn">Learn</Link>
-            <Link to="/posts/interview">Interview</Link>
-            <Link to="/posts/project">Project</Link>
-            <Link to="/posts/general">General</Link>
-            <Link to="/login">Login</Link>
-            <Link to="/register">Register</Link>
-            <Link to="/my-account">My Account</Link>
-            <Link to="/about-us">About Us</Link>
-        </nav>
+        <>
+            <AppBar position="fixed">
+                <Container maxWidth="xl">
+                    <Toolbar
+                        disableGutters
+                        sx={style.flexSpace.between}
+                    >
+                        <Typography
+                            variant="h6"
+                            noWrap
+                            component="a"
+                            href="/"
+                            sx={{...style.typography, ...style.typography.medium}}
+                        >
+                            DevPrep + LOGO
+                        </Typography>
+                        <Typography
+                            variant="h6"
+                            noWrap
+                            component="a"   
+                            href="/"
+                            sx={{...style.typography, ...style.typography.small}}
+                        >
+                            LOGO
+                        </Typography>
+                        
+                        <SmallView>
```

Comment:
As mentioned above, the functionality of SmallView is better to be implemented in the styles object. Rather than SmallView implementing styled components you can make this a generic container component like <Box> with the styling added to the styles object.